cirq-core/cirq/devices/grid_qubit.py

```diff
@@ -205,19 +205,25 @@ def rect(
 
     @staticmethod
     def from_diagram(diagram: str, dimension: int) -> List['GridQid']:
-        r"""Parse ASCII art device layout into info about qids and
-        connectivity. As an example, the below diagram will create a list of
-        GridQid in a pyramid structure.
+        r"""Parse ASCII art device layout into a device.
+
+        As an example, the below diagram will create a list of GridQid in a
+        pyramid structure.
+
+
+        ```
         ---A---
         --AAA--
         -AAAAA-
         AAAAAAA
+        ```
 
         You can use any character other than a hyphen to mark a qid. As an
```

Comment:
This is incorrect, as noted in the "Args: diagram" section below. Periods and spaces are also treated as non-Qid positions, and non-alphanumeric characters produce an error.